tools/src/main/java/org/apache/kafka/trogdor/fault/FaultSet.java

```diff
@@ -0,0 +1,144 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.kafka.trogdor.fault;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.util.Iterator;
+import java.util.Map;
+import java.util.NoSuchElementException;
+import java.util.TreeMap;
+
+public class FaultSet {
+    private static final Logger log = LoggerFactory.getLogger(FaultSet.class);
+
+    private final static long NS_PER_MS = 1000000L;
+
+    /**
+     * Maps fault start times in nanoseconds to faults.
+     */
+    private final TreeMap<Long, Fault> byStart = new TreeMap<Long, Fault>();
+
+    /**
+     * Maps fault end times in nanoseconds to faults.
+     */
+    private final TreeMap<Long, Fault> byEnd = new TreeMap<Long, Fault>();
+
+    /**
+     * Return an iterator that iterates over the fault set in start time order.
+     */
+    public FaultSetIterator iterateByStart() {
+        return new FaultSetIterator(byStart);
+    }
+
+    /**
+     * Return an iterator that iterates over the fault set in end time order.
+     */
+    public FaultSetIterator iterateByEnd() {
+        return new FaultSetIterator(byEnd);
+    }
+
+    /**
+     * Add a new fault to the FaultSet.
+     */
+    public void add(Fault fault) {
+        insertUnique(byStart, fault.spec().startMs() * NS_PER_MS, fault);
+        long endMs = fault.spec().startMs() + fault.spec().durationMs();
+        insertUnique(byEnd, endMs * NS_PER_MS, fault);
+    }
+
+    /**
+     * Insert a new fault to a TreeMap.
```

Comment:
Can you add a comment to say that the fault is stored in the next available key?